src/bidiMapper/domains/script/RealmStorage.ts

```diff
@@ -62,13 +63,21 @@ export class RealmStorage {
       }
       if (
         filter.browsingContextId !== undefined &&
-        filter.browsingContextId !== realm.browsingContextId
+        (!(realm instanceof WindowRealm) ||
+          filter.browsingContextId !== realm.browsingContext.id)
       ) {
         return false;
       }
       if (
         filter.navigableId !== undefined &&
-        filter.navigableId !== realm.navigableId
+        (!(realm instanceof WindowRealm) ||
+          filter.navigableId !== realm.browsingContext.navigableId)
```

Comment:
A list of `associatedNavigables` can be a part of `realm` as well.

Reply:
Actually, as long as `filter.navigableId` is not used anywhere, it can be simply removed.

Reply:
> Actually, as long as `filter.navigableId` is not used anywhere, it can be simply removed.

Seems like it!